src/hosting-node.js

```diff
@@ -51,59 +52,82 @@ async function getBody(url, userAgentIdentifier) {
 /**
  * Check if a domain is hosted by a green web host.
  * @param {string|array} domain - The domain to check, or an array of domains to be checked.
- * @param {object} db - Optional. A database object to use for lookups.
- * @param {string} userAgentIdentifier - Optional. The app, site, or organisation that is making the request.
- * @returns {boolean|array} - A boolean if a string was provided, or an array of booleans if an array of domains was provided.
+ * @param {string[] | DomainCheckOptions} optionsOrDb - Optional. An object of domain check options, or a database list to use for lookups.
+ * @param {string | DomainCheckOptions} optionsOrAgentId - Optional. An object of domain check options, or a string
+ *   representing the app, site, or organisation that is making the request.
+ * @returns - A boolean if a string was provided, or an array of booleans if an array of domains was provided.
+ *   if a string was provided for `domain`: a boolean indicating whether the domain is hosted by a green web host if `options.verbose` is false,
+ *     otherwise an object representing the domain host information.
+ *   if an array was provided for `domain`: an array of domains that are hosted by a green web host if `options.verbose` is false,
+ *     otherwise a dictionary of domain to host information.
  */
 
-function check(domain, db, userAgentIdentifier) {
+function check(domain, optionsOrDb, optionsOrAgentId) {
```

Comment:
@sfishel18 to move towards consistency between the node and api versions of the hosting check code, let's make _only_ the 2nd parameter the options object.

That is to say, the 2nd parameter could be a string (path to a database file) or an object (options). The 3rd parameter can remain as is (`userAgentIdentifier`). In our docs, we'll guide everyone to use the options object as the recommended way for passing in both the database and userAgentIndentifier. In that way, we can slowly deprecate the third option and move towards simplifying the code across both api and node versions. 

I'll make these changes, but capturing a comment to make my thoughts transparent.

Reply:
@fershad good call, makes sense to me!